browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -56,16 +60,88 @@ void OnShouldBlockAdResult(const ResponseCallback& next_callback,
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
```

Comment:
we should really be checking `g_brave_browser_process->IsShuttingDown()` here. This isn't new with your changes, but I think it's safer for the existing code. wdyt @iefremov ? I think next_callback.run could potentially crash here otherwise?

Reply:
actually this has the same problem as https://github.com/brave/brave-core/pull/6640/files#r492163653 I think because the browser process may have already been destroyed. Need to think about the best way to handle this

Reply:
`KeepAliveRegistry::GetInstance()->IsShuttingDown()` maybe?

Reply:
g_browser_process is destroyed at the end of `ShutdownThreadsAndCleanUp` so maybe this is safe as-is

Reply:
I think we are fine, until we don't hold any potentially dangling references - TaskRunners clean up their task queues on destruction. `next_callback` holds weak pointer to `BraveRequestHandler, so this is also safe.